### PR TITLE
Add summarize_results util

### PR DESCRIPTION
**Pull request by garretrieger**
This utility can pull specific information from the result proto. For example can print specific distributions or print only the total costs for each method.

### CODE REVIEW

File: analysis/analyzer.py [hunk 1]
```diff
@@ -31,6 +33,9 @@
     "Directory which contains all font's to be used in the analysis.")
 flags.mark_flag_as_required("font_directory")
 
+flags.DEFINE_bool("output_binary", False,
+                  "If true outputs the results in binary format.")
```
Comment (rsheeter) [inline analysis/analyzer.py]: suggest specifying what format, e.g. a binary X proto

File: tools/summarize_results.py [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+    input_file_contents = read_input(FLAGS.input_file)
+    text_format.Merge(input_file_contents, result_proto)
+  else:
+    input_file_contents = read_binary_input(FLAGS.input_file)
+    result_proto.ParseFromString(input_file_contents)
+
+  if len(argv) < 2:
+    return print_usage()
+
+  mode = argv[1]
```
Comment (rsheeter) [inline tools/summarize_results.py:73]: use a flag?

File: tools/summarize_results.py [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+  if mode == "cost_summary":
+    print_cost_summary(result_proto)
+    return 0
+
+  if mode in ("request_size_distribution",
```
Comment (rsheeter) [inline tools/summarize_results.py]: Suggest considering use of a map name : function plus flags rather than raw argv. Function would return a dist_proto.

Comment (rsheeter) [reply tools/summarize_results.py]: (assuming one can express cost_summary so?)

File: tools/summarize_results.py [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+    dist_proto = (network_proto.request_latency_distribution
+                  if mode == "latency_distribution" else
+                  network_proto.cost_distribution)
+    print_distribution(dist_proto)
+    return 0
+
+  return print_usage()
+
+
+def find_method_result(method, result_proto):
```
Comment (rsheeter) [inline tools/summarize_results.py:107]: suggest find should get the matching list and fail if len != 1 rather than just taking the first.

**Review by rsheeter — APPROVED** [review]

File: tools/summarize_results.py (hunk 2 above)
Comment (garretrieger) [reply tools/summarize_results.py:73]: Prefer to keep the mode stuff just as argv and it makes the command line less verbose.

File: tools/summarize_results.py (hunk 3 above)
Comment (garretrieger) [reply tools/summarize_results.py]: Switched to using a map of functions.

**Pull request merged**